connectivity/tests/host.go

```diff
@@ -88,21 +96,31 @@ func (s *podToHostPort) Run(ctx context.Context, t *check.Test) {
 		for _, echo := range ct.EchoPods() {
 			echo := echo // copy to avoid memory aliasing when using reference
 
-			baseURL := fmt.Sprintf("%s://%s:%d%s", echo.Scheme(), echo.Pod.Status.HostIP, check.EchoServerHostPort, echo.Path())
-			ep := check.HTTPEndpoint(echo.Name(), baseURL)
-			t.NewAction(s, fmt.Sprintf("curl-%d", i), &client, ep, check.IPFamilyAny).Run(func(a *check.Action) {
-				a.ExecInPod(ctx, ct.CurlCommand(ep, check.IPFamilyAny))
-
-				a.ValidateFlows(ctx, client, a.GetEgressRequirements(check.FlowParameters{
-					// Because the HostPort request is NATed, we might only
-					// observe flows after DNAT has been applied (e.g. by
-					// HostReachableServices),
-					AltDstIP:   echo.Address(check.IPFamilyAny),
-					AltDstPort: echo.Port(),
-				}))
-			})
+			var ipFamStr string
 
-			i++
+			t.ForEachIPFamily(func(ipFam check.IPFamily) {
+				if ipFam == check.IPFamilyV4 {
+					ipFamStr = "v4"
+				} else if ipFam == check.IPFamilyV6 {
+					ipFamStr = "v6"
+				}
+
+				baseURL := fmt.Sprintf("%s://%s:%d%s", echo.Scheme(), echo.Pod.Status.HostIP, check.EchoServerHostPort, echo.Path())
+				ep := check.HTTPEndpoint(echo.Name(), baseURL)
+				t.NewAction(s, fmt.Sprintf("curl-%s-%d", ipFamStr, i), &client, ep, check.IPFamilyAny).Run(func(a *check.Action) {
```

Comment:
There are many other places in the code were we construct the `curl-%d` string that also need to be extended with the IP family (for example, pod.go).

Reply:
Okay, that means, I need to make the following changes to [`pod.go`](https://github.com/cilium/cilium-cli/blob/master/connectivity/tests/pod.go) & [`service.go`](https://github.com/cilium/cilium-cli/blob/master/connectivity/tests/service.go) too, correct?

Reply:
Yes, basically all occurrences of ForEachIPFamily should be checked if we have different code based on the IPFamily and annotate the action name accordingly.